Add "Enabling Collaborative Editing" Yjs guide to the docs

CTNicholas opened the pull request:
Hey there!

I've written up a guide called "Enabling Collaborative Editing"—this takes you through setting up a multiplayer Slate editor using Yjs and slate-yjs. It also provides an example of how to set up a Liveblocks provider, and mentions our new package as a plugin. I've done my best to copy your writing and code style, but let me know if you need any changes! I'm not sure how to run a preview for the docs, so you'll have to try it out.

Thanks for building Slate, it's such a helpful tool.

dylans commented on docs/walkthroughs/07-enabling-collaborative-editing.md:
```diff
@@ -0,0 +1,397 @@
+That's all you need to attach Yjs to Slate!
+
+Let's look at a real-world example of setting up Yjs—here's a code snippet for setting up
+a [Liveblocks provider](https://liveblocks.io/docs/get-started/yjs-slate-react). Liveblocks uses the concept of rooms,
```
I would like to see a short disclaimer along the lines of (but maybe something better than my draft):

"Liveblocks is a commercial service independent of the Slate project. It is not required for enabling collaboration with Slate or Yjs, but may be a convenient option depending on your needs"

CTNicholas replied:
Good idea! I'll link to other providers too.

dylans commented on docs/general/resources.md:
```diff
@@ -53,15 +58,21 @@ These products use Slate, and can give you an idea of what's possible:
+  light-weight medium-style editor with no editor chrome.
+- [React Page](https://github.com/react-page/react-page) is a self-contained, customizable inline WYSIWYG editor
+  library.
+- [Slate Plugins Next](https://github.com/zbeyens/slate-plugins-next) provides an editor with configurable and
```
Not really relevant to your change, but might as well since I noticed how old this link is!

```suggestion
- [Plate (Plugins for Slate)](https://github.com/udecode/plate) provides an editor with configurable and
```

CTNicholas replied:
Got it!

CTNicholas commented:
Hey, thanks for reviewing this @dylans! I've updated the Plate link, added the text you recommend, and also added a link to other providers. Would you mind reviewing the changes I've made?
